include/bluetooth/bluetooth.h

```diff
@@ -497,6 +503,26 @@ int bt_le_scan_start(const struct bt_le_scan_param *param, bt_le_scan_cb_t cb);
  */
 int bt_le_scan_stop(void);
 
+/** @brief Register a scanner packet listener.
+ *
+ *  Adds the listener to the list of scanner packet callbacks. The listeners
+ *  are called on every received scanner packet, regardless of what initiated
+ *  the scan.
+ *
+ *  @param listener Listener to register. Must point to static memory.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_scan_listener_add(struct bt_le_scan_listener *listener);
```

Comment:
Rename to bt_le_scan_cb_register, given that we might want to have more scan callbacks?

This would also be in line with my current AE plans to have a bt_le_adv_cb_register for advertiser callbacks.

Both AE scanner and advertiser has a duration parameter which can be used to control the time or number of packets to send/receive.